common:
    lid_topic:  "/livox/lidar"
    imu_topic:  "/livox/imu"
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible
    time_offset_lidar_to_imu: 0.0 # Time offset between lidar and IMU calibrated by other algorithms, e.g. LI-Init (can be found in README).
                                  # This param will take effect no matter what time_sync_en is. So if the time offset is not known exactly, please set as 0.0

preprocess:
    lidar_type: 2                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR, 
    scan_line: 64
    scan_rate: 10                # only need to be set for velodyne, unit: Hz,
    timestamp_unit: 2            # the unit of time/t field in the PointCloud2 rostopic: 0-second, 1-milisecond, 2-microsecond, 3-nanosecond.
    blind: 16

mapping:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree:    180
    det_range:     100.0
    extrinsic_est_en:  false      # true: enable the online estimation of IMU-LiDAR extrinsic,
    extrinsic_T: [-8.086759e-01, 3.195559e-01, -7.997231e-01]
    extrinsic_R: [ 9.999976e-01, 7.553071e-04, -2.035826e-03,
                  -7.854027e-04, 9.998898e-01, -1.482298e-02,
                  2.024406e-03, 1.482454e-02, 9.998881e-01]

publish:
    dense_publish_en: true
    path_en: true
    scan_bodyframe_pub_en: true
    scan_publish_en: true
    debug_circular_scan_publish_en: true
    debug_circular_cloud_radius: 2.0

pcd_save:
    pcd_save_en: true
    interval: -1                 # how many LiDAR frames saved in each pcd file; 
                                 # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
wxx:
  # Lidar_Odom = Lidar_wrt_Body_R*(Body_Odom + Lidar_wrt_Body_T)
  Lidar_wrt_Body_R:
    [  1, 0, 0,
       0, 1, 0,
       0, 0, 1 ]
  Lidar_wrt_Body_T:
  - 0
  - 0
  - 0
  initial_map_from_pcd: false
  map_incremental: true
  initial_map_pcd_name: origin.pcd
  save_new_map_to_pcd: true
  new_map_pcd_name: new.pcd
